### DISCUSSION TITLE
32GB is not that much

### DISCUSSION THREAD

**Discussion by fungiboletus** [community/community#3785] [Codespaces]
I have used the same codespace for some days and today VS code failed to start because it ran out of storage while trying to upload the new VS Code remote server.

I then checked my disk usage, and I surely had 0 bytes left.

While 32GB may seem enough for a quick pull request, once you start using a codespace for some days you accumulate a lot of crap everywhere. Between the base system size, the wasted space because the block size of the file system is 4096 bytes, which wastes 10GB in my case, Docker containers and the many layers and logs, NPM packages, VSCode Remote cache, VSCode extensions caches, it's soon not enough storage for my 100kb of code.

I could destroy my codespace regularly or but it's annoying and time consuming. I think 48 or 64GB would be a __LOT__ more convenient. I already use the premium VM.

**Comment by TechStudent10** [comment]
I don't even have access to Codespaces after like 3 weeks soooooooooooooooooooooooooooooooooooooooo... 🤷🏾‍♂️

**Reply by fungiboletus** [reply]
You applied for early access pretty late. I was a Visual Studio Codespaces user before so I got access quickly.

**Reply by austinsonger** [reply]
@TechStudent11 **FYI** I signed up on the waiting list like 11 months ago so good luck.

**Reply by TechStudent10** [reply]
@austinsonger oh no...

**Comment by 2percentsilk** [comment]
Hi @fungiboletus, 

Thanks for this feedback - we're looking into how we can address this problem as you're not the only one who's reported this! 

A few follow-up questions from the team as we think about the best way to solve this problem:
1. Are you using the default image or a custom container?
2. Have you rebuilt that codespace in the last few days? - There's a known issue with the default image and rebuild that causes resource starvation that we're fixing now.
3. How long does it take for you to create a new codespace? How long would you like it to take if you were creating them often?
4. If the repo you're using is public, would you mind sharing a link to it?

**Reply by fungiboletus** [reply]
Hei @2percentsilk 

I'm happy to read that you will address the problem 😀

1. I'm using the default image so far, which is very convenient because it's all included. I'm considering to have a customer container to save some space if necessary. I don't need DotNet which is pretty big for example, but I do use NodeJS, Python, Java, or Golang.
2. Yes I actually did rebuild my codespace the 6th of April. Should I rebuild it again, or create a new Codespace, or wait ?
3. It's not very long for me, perhaps 30 minutes. I have to import some settings and secrets. Since I cannot create a Codespace from a private repository (yet), I import my setup manually instead of putting everything in my repository. Once I can upload a script to configure my codespace without it being public, it will take only a few minutes.
4. Since it has to be public (for me), I use an empty repository : https://github.com/fungiboletus/codespace Then I clone my private repositories using `gh repo clone` after I used `gh login`. I also have to do `export GITHUB_TOKEN=""` first. I know my flow is not what it's supposed to be, but I haven't found a better solution to have multiple private repositories in my codespace. My company doesn't use a monorepo configuration so I need to have a few repositories on the same machine. It's also why I use a lot of storage, as I have cache for NPM, cache from Maven, cache from Go, etc…

**Reply by sr229** [reply]
I think that's where the problem resides, Codespaces is for one repository only. While yes, it's nice to have a cloud IDE, it's tied to one repo and you're expected to use it on one repo AFAIK. I don't think it's wise to cram **all of your projects**, when the codespace is dedicated to one project only. You may have to change your workflow for that purpose.

Codespaces should be not tied to one repository IMHO, similar to VS Codespaces, if we are ever going to increase compute capacity.

**Comment by numbsafari** [comment]
Here is a repository that is basically empty: https://github.com/numbsafari/empty

If you launch a Codespace on that, this is where you start:

```
@numbsafari ➜ /workspaces/empty (main) $ df -h
Filesystem      Size  Used Avail Use% Mounted on
overlay          32G   11G   20G  34% /
tmpfs            64M     0   64M   0% /dev
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
shm              64M  8.0K   64M   1% /dev/shm
/dev/sda1        29G   10G   19G  35% /usr/sbin/docker-init
/dev/loop0       32G   11G   20G  34% /workspaces
/dev/sdb1        32G   49M   30G   1% /tmp
```

11GB of space is used just for all the "stuff" needed to bootstrap a codespace. As noted by others, things quickly break down once you start actually using a space for work.

**Comment by jeremysf** [comment]
+1 to this. I can't use codespaces to build and iterate docker images as I am constantly running out of disk space.

**Reply by brannondorsey** [reply]
+1 to this use case. I'm unable to build and develop inside my own Docker containers in Codespaces because of this limitation, something which is a requirement for my repository. I would be happy to pay for more disk space usage if it was supported.

**Comment by topazus** [comment]
I agree with the argument, the default 32GB may be too small. By the way, I'm using the Fedora docker image.

**Comment by Miladew** [comment]
> I have used the same codespace for some days and today VS code failed to start because it ran out of storage while trying to upload the new VS Code remote server.
> 
> I then checked my disk usage, and I surely had 0 bytes left.
> 
> While 32GB may seem enough for a quick pull request, once you start using a codespace for some days you accumulate a lot of crap everywhere. Between the base system size, the wasted space because the block size of the file system is 4096 bytes, which wastes 10GB in my case, Docker containers and the many layers and logs, NPM packages, VSCode Remote cache, VSCode extensions caches, it's soon not enough storage for my 100kb of code.
> 
> I could destroy my codespace regularly or but it's annoying and time consuming. I think 48 or 64GB would be a **LOT** more convenient. I already use the premium VM.

😍

**Comment by Miladew** [comment]
> I have used the same codespace for some days and today VS code failed to start because it ran out of storage while trying to upload the new VS Code remote server.
> 
> I then checked my disk usage, and I surely had 0 bytes left.
> 
> While 32GB may seem enough for a quick pull request, once you start using a codespace for some days you accumulate a lot of crap everywhere. Between the base system size, the wasted space because the block size of the file system is 4096 bytes, which wastes 10GB in my case, Docker containers and the many layers and logs, NPM packages, VSCode Remote cache, VSCode extensions caches, it's soon not enough storage for my 100kb of code.
> 
> I could destroy my codespace regularly or but it's annoying and time consuming. I think 48 or 64GB would be a **LOT** more convenient. I already use the premium VM.

😍

**Comment by smanchester-avantus** [comment]
Reviving this thread, it seems like a fairly trivial feature to build slightly more configurability where we can have more disk space if needed.  Is this being considered at all?  If yes, can you link to the feature request, and if no can you explain the reasoning? Each of the above complains really resonate for me and my team.

**Reply by nick-youngblut** [reply]
In regards to machine type configurability, it would also be helpful to have some options for high-memory, low-core machines (e.g., 64 Gb RAM, but only 4-8 cores). Bioinformatics software development often requires large amounts of memory, but not necessarily many cores. 

I also am constantly run out of disk space, and using s3fs causes all kinds of issues with slow or non-responsive codespace images.

**Comment by elisim** [comment]
Is there any updates on this issue from the github team?
working on https://github.com/huggingface/transformers and the disk almost full after installing the dev dependencies

**Comment by steffennilsen** [comment]
Ive experienced constant need to restart and clean my codespace today after working on a tiny 56k CLI tool, once docker is involved the GBs fly quickly

**Comment by Hyphonical** [comment]
Is this still being worked on?

**Comment by Ronnie-personal** [comment]
My codespace runs out of disk space. The project that I'm working on requires at least 60GB disk space, but the Codespace root volume is only 32GB.
I tried to change machine type, but the only available options are all 32GB storage.

I'm willing to pay for more disk if there is an option to upgrade to 64 or 128GB storage.
@2percentsilk Appreciate your help if you have any recommendations!

**Reply by husshuss179** [reply]
found a fix?

**Reply by sriluxman** [reply]
Is there a fix?

**Reply by numbsafari** [reply]
GCP dev workstations allow you to configure 10, 50, 100, 200, 500, or 1000 GB storage devices for instances.

That's the solution I'm looking at.

**Comment by smanchester-avantus** [comment]
Obligatory XKCD reference at this point: https://m.xkcd.com/2881/

**Reply by rayjlinden** [reply]
So you want to get a beach house this weekend?

**Comment by dbold** [comment]
128GB also isn't that much for some projects. Codespaces needs a 512GB or even 1TB option.

This little machine type popup https://docs.github.com/en/codespaces/customizing-your-codespace/changing-the-machine-type-for-your-codespace really needs to add an option where the exact `devcontainer.json` `hostRequirements` is used to create a machine.

**Comment by kingdudely** [comment]
Type in `lsblk` in the terminal and you will find a 512 GB hard drive but IDK how to use it
